dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/map-to-fee-record-utilisation.test.ts

```diff
@@ -0,0 +1,198 @@
+      const coverPercentage = 80;
+      const facilityId = '12345678';
+
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(aUtilisationReport()).withFacilityId(facilityId).build();
+
+      const tfmFacility = aTfmFacility();
+      tfmFacility.facilitySnapshot.ukefFacilityId = facilityId;
+      tfmFacility.facilitySnapshot.coverPercentage = coverPercentage;
+
+      // Act
+      const feeRecordUtilisation = mapToFeeRecordUtilisation(feeRecord, tfmFacility);
+
+      // Assert
+      expect(feeRecordUtilisation.coverPercentage).toEqual(coverPercentage);
+    });
+
+    it('should set the value to the corresponding field in the facility snapshot when there are no amendments to the value', () => {
+      // Arrange
+      const value = 200000;
+      const facilityId = '12345678';
+
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(aUtilisationReport()).withFacilityId(facilityId).build();
+
+      const tfmFacility = aTfmFacility();
+      tfmFacility.facilitySnapshot.ukefFacilityId = facilityId;
+      tfmFacility.facilitySnapshot.value = value;
+
+      const getLatestCompletedAmendmentFacilityValueSpy = jest.spyOn(helpers, 'getLatestCompletedAmendmentToFacilityValue').mockReturnValue(undefined);
+
+      // Act
+      const feeRecordUtilisation = mapToFeeRecordUtilisation(feeRecord, tfmFacility);
+
+      // Assert
+      expect(getLatestCompletedAmendmentFacilityValueSpy).toHaveBeenCalledTimes(1);
+      expect(getLatestCompletedAmendmentFacilityValueSpy).toHaveBeenCalledWith(tfmFacility);
+      expect(feeRecordUtilisation.value).toEqual(value);
+    });
+
+    it('should set the value to the latest amended value when there is an amendement to the value', () => {
+      // Arrange
+      const originalValue = 200000;
+      const amendedValue = 300000;
+      const facilityId = '12345678';
+
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(aUtilisationReport()).withFacilityId(facilityId).build();
+
+      const tfmFacility = aTfmFacility();
+      tfmFacility.facilitySnapshot.ukefFacilityId = facilityId;
+      tfmFacility.facilitySnapshot.value = originalValue;
+
+      const getLatestCompletedAmendmentFacilityValueSpy = jest.spyOn(helpers, 'getLatestCompletedAmendmentToFacilityValue').mockReturnValue(amendedValue);
+
+      // Act
+      const feeRecordUtilisation = mapToFeeRecordUtilisation(feeRecord, tfmFacility);
+
+      // Assert
+      expect(getLatestCompletedAmendmentFacilityValueSpy).toHaveBeenCalledTimes(1);
+      expect(getLatestCompletedAmendmentFacilityValueSpy).toHaveBeenCalledWith(tfmFacility);
+      expect(feeRecordUtilisation.value).toEqual(amendedValue);
+    });
+
+    it('should calculate the exposure from the utilisation and cover percentage', () => {
+      // Arrange
+      const facilityId = '12345678';
+      const utilisation = 1000;
+      const coverPercentage = 60;
+      // The expected exposure = 1000 * 60 / 100 = 600
+      const expectedExposure = 600;
```

Comment:
this value should come from `calculateExposure`

Reply:
I think it is worth having this test here, I'm only exporting and unit testing the exposure calculation separately because we have to write tests for each function.

Reply:
Correct and the tests for `calculateExposure` will fail if something was to go wrong with the function.  Hard coding this could make it harder to debug a test failure if something was to change with `calculateExposure` so this should not be hard coded.  Spoke to fellow devs about this @ttbarnes

Reply:
I disagree, but I have changed it